handler/enqueue_annotation_test.go

```diff
@@ -0,0 +1,432 @@
+				Object: pod,
+				Meta:   pod.GetObjectMeta(),
+			}
+			instance.Generic(evt, q)
+			Expect(q.Len()).To(Equal(1))
+
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: podOwner.Namespace,
+					Name:      podOwner.Name,
+				},
+			}))
+		})
+
+		It("should not enqueue a request if there are no annotations matching with the object", func() {
+			var repl *appsv1.ReplicaSet
+
+			repl = &appsv1.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "faz",
+				},
+			}
+
+			evt := event.CreateEvent{
+				Object: repl,
+				Meta:   repl.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(0))
+		})
+
+		It("should not enqueue a Request if there is no Namespace and name annotation matching the specified object", func() {
+			var repl *appsv1.ReplicaSet
+
+			repl = &appsv1.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "faz",
+					Annotations: map[string]string{
+						TypeAnnotation: schema.GroupKind{Group: "Pods", Kind: "core"}.String(),
+					},
+				},
+			}
+
+			evt := event.CreateEvent{
+				Object: repl,
+				Meta:   repl.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(0))
+		})
+
+		It("should not enqueue a Request if there is no TypeAnnotation matching Group and Kind", func() {
+			var repl *appsv1.ReplicaSet
+
+			repl = &appsv1.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "faz",
+
+					Annotations: map[string]string{
+						NamespacedNameAnnotation: "AppService",
+					},
+				},
+			}
+
+			evt := event.CreateEvent{
+				Object: repl,
+				Meta:   repl.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(0))
+		})
+
+		It("should enqueue a Request if there are no Namespace annotation matching for the object", func() {
+			var repl *appsv1.ReplicaSet
+
+			repl = &appsv1.ReplicaSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "faz",
+					Annotations: map[string]string{
+						NamespacedNameAnnotation: "AppService",
+						TypeAnnotation:           schema.GroupKind{Group: "Pods", Kind: "core"}.String(),
+					},
+				},
+			}
+
+			evt := event.CreateEvent{
+				Object: repl,
+				Meta:   repl.GetObjectMeta(),
+			}
+
+			instance.Create(evt, q)
+			Expect(q.Len()).To(Equal(1))
+
+			i, _ := q.Get()
+			Expect(i).To(Equal(reconcile.Request{
+				NamespacedName: types.NamespacedName{Namespace: "", Name: "AppService"}}))
+		})
+
+		It("should enqueue a Request for a object that is cluster scoped which has the annotations", func() {
+			var nd *corev1.Node
+
+			nd = &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "node-1",
+					Annotations: map[string]string{
+						NamespacedNameAnnotation: "myapp",
+						TypeAnnotation:           schema.GroupKind{Group: "ReplicaSet", Kind: "apps"}.String(),
```

Comment:
This is also backwards:
```suggestion
						TypeAnnotation:           schema.GroupKind{Group: "apps", Kind: "ReplicaSet"}.String(),
```

Ditto in other places...